app/browser/api/ledger.js

```diff
@@ -2718,22 +2718,43 @@ const onPublisherTimestamp = (state, oldTimestamp, newTimestamp) => {
 }
 
 const checkReferralActivity = (state) => {
+  console.log('checkReferralActivity')
   const downloadId = updateState.getUpdateProp(state, 'referralDownloadId')
 
   if (!downloadId) {
     updater.checkForUpdate(false, true)
-    return
+    return state
   }
 
-  module.exports.roundtrip({
-    server: referralServer,
-    method: 'PUT',
-    path: '/promo/activity',
-    payload: {
-      download_id: downloadId,
-      api_key: referralAPI
-    }
-  }, {}, activityRoundTrip)
+  const timestamp = updateState.getUpdateProp(state, 'referralAttemptTimestamp') || 0
+  const count = updateState.getUpdateProp(state, 'referralAttemptCount') || 0
+
+  console.log('counter', count)
+
+  if (count >= 5) {
+    console.log('I gave up :(')
```

Comment:
I know you'll remove the `console.log` lines... but I think this would be good to log with `LEDGER_VERBOSE` 😄

Reply:
added